sdk/attestation/Azure.Security.Attestation/src/AttestationToken.cs

```diff
@@ -143,43 +202,390 @@ protected AttestationToken()
 
         /// <summary>
         /// Validate a JSON Web Token returned by the MAA.
+        /// <para/>
+        /// If the caller provides a set of signers, than that set of signers will be used as the complete set of candidates for signing.
+        /// If the caller does not provide a set of signers, then the <see cref="ValidateTokenAsync(TokenValidationOptions, IReadOnlyList{AttestationSigner}, CancellationToken)"/>
+        /// API will a set of callers derived from the contents of the attestation token.
         /// </summary>
+        /// <param name="options">Options used while validating the attestation token.</param>
         /// <param name="attestationSigningCertificates">Signing Certificates used to validate the token.</param>
-        /// <param name="validationCallback">User provided callback which allows the customer to validate the token.</param>
-        /// <returns></returns>
-        public virtual bool ValidateToken(IReadOnlyList<AttestationSigner> attestationSigningCertificates, Func<AttestationToken, AttestationSigner, bool> validationCallback = default)
+        /// <param name="cancellationToken">Token used to cancel this operation if necessary.</param>
+        /// <returns>true if the token was valid, false otherwise.</returns>
+        /// <exception cref="ArgumentNullException">Thrown if the signing certificates provided are invalid.</exception>
+        /// <exception cref="Exception">Thrown if validation fails.</exception>
+        public virtual async Task<bool> ValidateTokenAsync(TokenValidationOptions options, IReadOnlyList<AttestationSigner> attestationSigningCertificates, CancellationToken cancellationToken = default)
+        {
+            // Early out if the caller doesn't want us to validate the token.
+            if (!options.ValidateToken)
+            {
+                return true;
+            }
+
+            // Before we waste a lot of time, see if the token is unsecured. If it is, then validation is simple.
+            if (Header.Algorithm.Equals("none", StringComparison.OrdinalIgnoreCase))
+            {
+                if (!ValidateCommonProperties(options))
+                {
+                    return false;
+                }
+                if (options.ValidationCallback != null)
+                {
+                    return options.ValidationCallback(this, null);
+                }
+                return true;
+            }
+
+            // This token is a secured attestation token. If the caller provided signing certificates, then
+            // they need to have provided at least one certificate.
+            if (attestationSigningCertificates != null)
+            {
+                Argument.AssertNotNull(attestationSigningCertificates[0], nameof(attestationSigningCertificates));
+            }
+
+            var possibleCertificates = await GetCandidateSigningCertificatesAsync(attestationSigningCertificates, cancellationToken).ConfigureAwait(false);
```

Comment:
Why is only async call in this method a call to an async method whose implementation is actually sync (i.e. implemented as `Task.FromResult`)?  As far as I can tell, this method will never asynchronously wait and will always return a completed task when called.